### PR TITLE
Simplify r_anal_callbacks_t

### PR DESCRIPTION
**Pull request by MaskRay**

### CODE REVIEW

File: libr/core/core.c [hunk 1]
```diff
@@ -73,7 +73,7 @@ static int on_fcn_delete (void *_anal, void* _user, RAnalFunction *fcn) {
 	return 0;
 }
 
-static int on_fcn_rename(void *_anal, void* _user, RAnalFunction *fcn, const char *oname) {
+static int on_fcn_rename(void* _user, RAnalFunction *fcn, const char *oname) {
 	RCore *core = (RCore*)_user;
```
Comment (radare) [inline libr/core/core.c]: anal != core this change is wrong

Comment (MaskRay) [reply libr/core/core.c]: It would break tests if this were wrong. no one uses the first argument.

**Comment by radare** [comment]
but callabcsk should honor the ->user parameter, this can be a different anal instance, or RCore like in this case. didnt checked who calls or uses this

**Comment by MaskRay** [comment]
I removed `RAnal*` not `RCore*`.

**Comment by MaskRay** [comment]
Then remove the two uncivilized macros.

File: libr/include/r_anal.h [hunk 2]
```diff
@@ -576,15 +576,12 @@ typedef struct r_anal_switch_obj_t {
...
-//struct r_anal_t*
-#define RANAL_BLOCK void*
-//struct r_anal_bb_t*
+struct r_anal_bb_t;
 typedef struct r_anal_callbacks_t {
-	int (*on_fcn_new) (RANAL, void *user, RAnalFunction *fcn);
-	int (*on_fcn_delete) (RANAL , void *user, RAnalFunction *fcn);
-	int (*on_fcn_rename) (RANAL, void *user, RAnalFunction *fcn, const char *oldname);
-	int (*on_fcn_bb_new) (RANAL, void *user, RAnalFunction *fcn, RANAL_BLOCK bb);
+	int (*on_fcn_new) (void *ana, void *user, RAnalFunction *fcn);
```
Comment (XVilka) [inline libr/include/r_anal.h]: Why not `void *anal`? Better to be consistent across the program.

Comment (MaskRay) [reply libr/include/r_anal.h]: Improve the naming BTW.

Comment (ret2libc) [reply libr/include/r_anal.h]: I think we don't want to follow IDA and prefer to have `anal`.

Comment (ret2libc) [reply libr/include/r_anal.h]: Also, why isn't this using `RAnal`/`struct r_anal_t` type instead of `void *`?

**Comment by MaskRay** [comment]
done

File: libr/include/r_anal.h [hunk 3]
```diff
@@ -576,15 +576,13 @@ typedef struct r_anal_switch_obj_t {
 	RList *cases;
 } RAnalSwitchOp;
 
-#define RANAL void*
-//struct r_anal_t*
-#define RANAL_BLOCK void*
-//struct r_anal_bb_t*
+struct r_anal_t;
```
Comment (radare) [inline libr/include/r_anal.h]: i think this was causing problems in some compilers, but it's green, so i guess it's fine now

**Pull request merged**